src/compile/data/assemble.ts

```diff
@@ -307,8 +312,8 @@ export function assembleFacetData(root: FacetNode): VgData[] {
  * @param  data array
  * @return modified data array
  */
-export function assembleData(dataCompomponent: DataComponent): VgData[] {
-  let roots: SourceNode[] = vals(dataCompomponent.sources);
+export function assembleData(model: Model, dataComponent: DataComponent): VgData[] {
```

Comment:
I'm not sure I understand what you're proposing. Would you rather the selection component be passed to `assembleData`. That seems awfully brittle, but I'm probably misunderstanding.